exercises/practice/leap/.approaches/if-expression/content.md

```diff
@@ -0,0 +1,53 @@
+# If expression
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+  let
+    divisibleBy number = 
+      modBy number year == 0 
+  in
+    if divisibleBy 400 then
+      True
+    
+    else if divisibleBy 100 then
+      False
+    
+    else 
+      divisibleBy 4
+```
+
+An [`if` expression][if-expression] (`if … then … else …`) is a compound expression that uses a test to determine which of two alternatives to evaluate to.
+Many other languages feature a similar construct, often termed 'ternary operator'.
+
+## In this approach
+
+This approach uses exactly two tests to determine whether a year is a leap year.
+
+The first test is for divisibility by 100.
```

Comment:
This doesn't match the actual code, since we start by checking 400.
From the description, the code would probably be (untested)
```elm
isLeapYear : Int -> Bool
isLeapYear year =
  let
    divisibleBy number = 
      modBy number year == 0 
  in
    if divisibleBy 100 then
      divisibleBy 400

    else
      divisibleBy 4
```
which only has one if statement. If you don't want to change the description, I would consider changing to this code (after double checking that it passes the tests).

Reply:
It does pass the tests, I'll use it ...